ledger/src/snapshot_utils.rs

```diff
@@ -270,10 +287,19 @@ where
         .ok_or_else(|| get_io_error("No snapshots found in snapshots directory"))?;
 
     // Rebuild the root bank
-    info!("Loading from {:?}", &root_paths.snapshot_file_path);
+    info!("Loading bank from {:?}", &root_paths.snapshot_file_path);
     let file = File::open(&root_paths.snapshot_file_path)?;
     let mut stream = BufReader::new(file);
-    let bank: Bank = deserialize_from(&mut stream)?;
+    let bank: Bank = match snapshot_version {
+        env!("CARGO_PKG_VERSION") => deserialize_from(&mut stream)?,
+        "v0.22.3" => {
+            let bank0223: solana_runtime::bank::LegacyBank0223 = deserialize_from(&mut stream)?;
+            bank0223.into()
+        }
+        _ => {
+            return Err(get_io_error("unsupported snapshot version"));
```

Comment:
nits: Care to include the actual version string? That extra effort will save us a ton in a debugging where third parties are often involved.

Reply:
for sure, done